src/comps/orth.cc

```diff
@@ -45,6 +65,33 @@ int CholQRQ<T>::cholqrq(
     return 0;
 }
 
+// -----------------------------------------------------------------------------
+/// Performs an unpivoted LU factorization. Outputs the L-factor only.
+/// Uses L-extraction routine and LASWP().
+///
+/// Templated for `float` and `double` types.
+///
+/// @param[in] m
+///     The number of rows in the matrix A.
+///
+/// @param[in] n
+///     The number of columns in the matrix A.
+///
+/// @param[in] A
+///     The m-by-n matrix, stored in a column-major format.
+///
+/// @param[in] ipiv
+///     Buffer for the pivot vector.
+///     
+/// @param[out] A
+///     Overwritten by the lower-triangular factor L with interchanged rows,
+///     L[ipiv,:].
```

Comment:
The semantics of a "stabilizer" require that the output matrix has the same range as the input matrix. So I think that in order to be consistent with this, you'd need to zero out the strict upper triangle of ``A`` before returning. I think you do this with a separate utility function somewhere ...